### DISCUSSION TITLE
`defaultProps` and `LayerExtension`

### DISCUSSION THREAD

**Discussion by ilan-gold** [visgl/deck.gl#6310] [Q&A]
Hello,

I have tried adding 
```javascript
class Extension extends deck.LayerExtension {
  draw(){
    console.log(this.props)
  }
}

Extension.defaultProps = {foo: 'bar'}
```
to a code pen to debug dether extension-specific `defaultProps` are used and I didn't see the `foo` prop logged out: https://codepen.io/ilan-gold/pen/JjyRbPe?editors=1112 (this is just a fork of one of you code pens with the above added).  I can see the other props logged but not this default prop.  I see the `defaultProps` used on some of the extensions implemented in this repo so I thought they functioned similarly to layer props.

**Comment by ilan-gold** [comment]
I have greatly simplified the codepen now.  Should be "bare minimum".

**Reply by jfuehner** [reply]
Hey @ilan-gold this definitely appears to be a known issue. If you look at [BrushingExtension](https://github.com/visgl/deck.gl/blob/master/modules/extensions/src/brushing/brushing.js) there is a reference to the `defaultProps` not being merged.

![image](https://user-images.githubusercontent.com/1843941/140832342-50bcdda6-7855-4f39-8431-aa58ab7c17f1.png)

The [ClipExtension](https://github.com/visgl/deck.gl/blob/master/modules/extensions/src/clip/clip.js) appears to directly reference its out defined `defaultProps` object within the `draw` function to most likely get around the limitation.

**Comment by ilan-gold** [comment]
Just to be clear, is this a bug or intended behavior?  If the former, we should open an issue and fix it!
